meson.build

```
@@ -0,0 +1,88 @@
+project(
+  'vstring',
+  'cpp',
+  version: '0.0.0999',
+  meson_version : '>= 0.48.0',
+  default_options: [
+    'warning_level=3',
+    'b_lto=true',
+    'b_pie=true'
+    ]
+  )
+
+cxx = meson.get_compiler('cpp')
+extra_flags = [
+  '-fno-common',
+  '-Wshadow',
+  '-Wformat-security',
+  '-Wformat-overflow=2',
+  '-Wdate-time',
+  '-D_FORTIFY_SOURCE=2',
+  '-DVSTRING_VERSION="@0@"'.format(meson.project_version())
+]
+
+foreach cxxflag : extra_flags
+  if cxx.has_argument(cxxflag)
+    add_project_arguments(cxxflag, language: 'cpp')
+  endif
+endforeach
```

```suggestion
add_project_arguments(cxx.get_supported_arguments(extra_flags), language: 'cpp')
```

Available since 0.43.0